bench_test.go

```diff
@@ -0,0 +1,251 @@
+package yarpc_test
+
+import (
+	"bytes"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/yarpc/yarpc-go"
+	"github.com/yarpc/yarpc-go/encoding/raw"
+	"github.com/yarpc/yarpc-go/transport"
+	yhttp "github.com/yarpc/yarpc-go/transport/http"
+	ytchannel "github.com/yarpc/yarpc-go/transport/tchannel"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	traw "github.com/uber/tchannel-go/raw"
+	"golang.org/x/net/context"
+	"golang.org/x/net/context/ctxhttp"
+)
+
+var _reqBody = []byte("hello")
+
+func yarpcEcho(reqMeta yarpc.ReqMeta, body []byte) ([]byte, yarpc.ResMeta, error) {
+	return body, yarpc.NewResMeta(reqMeta.Context()).Headers(reqMeta.Headers()), nil
+}
+
+func httpEcho(t testing.TB) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+		hs := w.Header()
+		for k, vs := range r.Header {
+			for _, v := range vs {
+				hs.Set(k, v)
+			}
+		}
+
+		_, err := io.Copy(w, r.Body)
+		assert.NoError(t, err, "failed to write HTTP response body")
+	}
+}
+
+type tchannelEcho struct{ t testing.TB }
+
+func (tchannelEcho) Handle(ctx context.Context, args *traw.Args) (*traw.Res, error) {
+	return &traw.Res{Arg2: args.Arg2, Arg3: args.Arg3}, nil
+}
+
+func (t tchannelEcho) OnError(ctx context.Context, err error) {
+	t.t.Fatalf("request failed: %v", err)
+}
+
+func withDispatcher(t testing.TB, cfg yarpc.Config, f func(yarpc.Dispatcher)) {
```

Comment:
It's not necessary IMO. Trivial for users to do `defer`. Usually they won't have multiple Dispatchers objects in play.